backend/modules/simulation/services/lifecycle_service.py

```
@@ -219,13 +222,43 @@ async def start_simulation(
+                from common.db.models import UserProgress as UP
+                user_progress = self.db.query(UP).filter(UP.id == progress_id).first()
+                user_progress.sandbox_id = sandbox_id
+                self.db.commit()
+                logger.info(f"[LIFECYCLE] Sandbox {sandbox_id} created for user {user_id}")
+                # Upload first scene's data files into the sandbox
+                if sandbox_id and getattr(first_scene, "scene_type", "conversation") == "code_challenge":
+                    scene_data_files = getattr(first_scene, "data_files", None)
+                    if scene_data_files:
+                        count = await sandbox_service.upload_scene_data_files(sandbox_id, scene_data_files)
+                        logger.info(f"[LIFECYCLE] Uploaded {count} data files to sandbox for scene {first_scene.id}")
+            except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)